Linkta-org/core
# #240 forgot/reset password

- **change navigation route to on google auth sign in**
- **add send password link functionality; add snackbar toast messages**


## Description
Fixes: #240 
This PR creates the password reset functionality for users created with email and passwords.  




Fixes # (issue)

## Type of change



## How Can this be tested? Testing Plan to review this PR


1.  Fire up the app and go to the `login` route and click on the `update` link next to forgot password. This takes you to the ‘forgotten’ route 
2. Enter a random email that doesn’t exist in our database.  See Snackbar notification for failing. 
3. Enter an email address that you have used to create a user in our App.  If you haven’t created one yet, go to create user and do so. 
4. Enter your email and see Snackbar messages for success and errors. 
5. Check your email for link and click on it.  This will take you back to the Linkta app to our `update-password` route.  
6. Type in a new password.  password must contain the follwoing: 1 uppercase letter, 1 lowercase letter, 1 number and 1 symbol.  Confirm your password in the Confirm password and submit.  
7. See SnackbarNotification for Success and be rerouted to `login` after 2.5 seconds

Currently there is no error handling for submitting an email that doesn’t exist in our database.  I tried using the `fetchSignInMethodsForEmail` - a firebase method intended for this, but this has been deprecated for for security  improvements around email enumeration. New projects created after 9/15/23 don’t have access to this function

[This article](https://medium.com/coffee-software/firebase-email-enumeration-enforcement-a-workaround-for-fetchsigninmethodsforemail-13a72a88f26e) suggests implementing this with a Firebase function.  I think this merits some looking into in a follow on ticket.  

## Checklist:



## Additional Information:
